lib/tools/apk-utils.js

```diff
@@ -498,19 +498,34 @@ apkUtilsMethods.setDeviceLocale = async function (locale) {
  * @return {boolean} If current locale is language and country as arguments, return true.
  */
 apkUtilsMethods.ensureCurrentLocale = async function (language, country) {
-  if (!_.isString(language) && !_.isString(country)) {
+  let hasLanguage = _.isString(language);
+  let hasCountry = _.isString(country);
+
+  if (!hasLanguage && !hasCountry) {
     log.warn('ensureCurrentLocale requires language or country');
     return false;
   }
 
-  const lowLanguage = language.toLowerCase();
-  const lowCountry = country.toLowerCase();
+  const lowLanguage = (language || '').toLowerCase();
+  const lowCountry = (country || '').toLowerCase();
 
   if (await this.getApiLevel() < 23) {
-    const curLanguage = (await this.getDeviceLanguage()).toLowerCase();
-    const curCountry = (await this.getDeviceCountry()).toLowerCase();
```

Comment:
I assume it's still ok to get these values before the if condition

Reply:
is it an expensive operation?

Reply:
It's just to reduce the number of calling `adb` commands and I also considered as you addressed. The operation cost isn't expensive, I believe...

Reply:
how about this?
```js
let curLanguage, curCountry;
if (hasLanguage) {
    curLanguage = (await this.getDeviceLanguage()).toLowerCase();
    if (!hasCountry && lowLanguage === curLanguage) {
        return true;
    }
}
if (hasCountry) {
    curCountry = (await this.getDeviceCountry()).toLowerCase();
    if (!hasLanguage && lowCountry === curCountry) {
        return true;
    }
}
if (lowLanguage === curLanguage && lowCountry === curCountry) {
    return true;
}
```

Reply:
👍 
looks good. thanks!